src/fu-engine.c

```diff
@@ -6119,6 +6156,14 @@ fu_engine_backend_device_added_cb (FuBackend *backend, FuDevice *device, FuEngin
 		g_debug ("%s added %s", fu_backend_get_name (backend), str);
 	}
 
+	/* engine safety checks */
+	if (!fu_engine_check_device_safe (self, device, &error_local)) {
```

Comment:
Isn't this quite a big change in behaviour? Before we wouldn't probe or setup the device as the plugin was disabled, and with this it seems we'd probe, setup and only ignore the device when it's being added. IIRC wasn't it the initialisation that caused the kernel to lockup? This seems like the sort of thing you want to do in `fu_engine_load_plugins` and set `FWUPD_PLUGIN_FLAG_DISABLED` with `FWUPD_PLUGIN_FLAG_KERNEL_TOO_OLD` or something.

Reply:
It should be before the plugin's probe function runs, look specifically it's before fu_device_probe runs.

I can certainly see the argument moving it to where you suggested though too.

Reply:
want me to have a go, or you already done? :)

Reply:
Go ahead :)

Reply:
Alternate proposal in https://github.com/fwupd/fwupd/pull/3453